src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/ExposureDetectionViewController.swift

```diff
@@ -262,11 +263,12 @@ extension ExposureDetectionViewController {
 // MARK: - CountdownTimerDelegate methods.
 
 extension ExposureDetectionViewController: CountdownTimerDelegate {
-	func update(time: String) {
+
+	func countdownTimer(_ timer: CountdownTimer, didUpdate time: String) {
 		self.updateCheckButton(time)
 	}
 
-	func done() {
+	func countdownTimer(_ timer: CountdownTimer, didEnd done: Bool) {
```

Comment:
same goes here, is it necessary to pass the timer object?

Reply:
the timer is not used here, however I am trying to stick to the apple style delegates as Hao proposed where you get passed a reference to the caller as first parameter.

Reply:
Same comment I gave before to Hao, what about the YAGNI principle?